Redesign "My videos/series/playlists" pages

owi92 opened the pull request:
This adds a new layout and some filtering options for the "My videos", "My series" and "My playlists" overview pages.

LukasKalbertodt commented on frontend/src/routes/manage/Series/index.tsx:
```diff
@@ -85,6 +81,7 @@ const query = graphql`
                     state
                     numVideos
                     thumbnailStack { thumbnails { url live audioOnly }}
+                    hostRealms { id }
```
That causes a 1+n query problem. TODO for myself: understand the rest of the PR and make suggestions how to fix.

owi92 replied:
I think that was a leftover or sth. Hard to say now. Anyway, it's not even needed here. -> removed

owi92 committed:
Make search input more configurable (height, icon, spinner size)

This changes makes the input reusable in different contexts
by adjusting the values of the addedi optional props.

owi92 committed:
Export `MenuItem` and adjust some props

Again, making the component reusable for other
contexts (soon to be used for table sorting menus).

owi92 committed:
Make `VideoShareButton` more flexible

This allows using the button without a timestamp
in different contexts.

owi92 committed:
Add `AccessIcon` component

This will show an icon to indicate the visibily stat
of a video or playlist. That visibility can either be
- "public', i.e. it can be viewed (read) by everyone (via
`ROLE_ANONYMOUS`),
- "shared", meaning that there is more than one person who
has read access for that event, or
- "private", in which case the item can only be read by a single
non-admin user.

There is also a tooltip for these icons with a short explaination.

The icon is to be placed in/above (above as "on top") the thumbnail.
Since we use different layouts for video- and playlist thumbnails
(series do get this indicator because their ACL meaning is kinda weird),
this `AccessIcon` component uses a specific placement for either, so
it is always shown at the top right of the thumbnail in the front (playlists
use a sort of staggered layout with three individual thumbnails).

This is too much text for a simple component, but whatever, nobody reads
these anyway, right?
Oh I forgot to mention: The icon also uses a drop shadow, so it works on
both dark and light backgrounds.

owi92 committed:
Extend backend table search filters

This adds filters for date and visibility (based on ACL)
and adds the description field to the text filter
(a single query now shows match results for both title
and description).
Also exposes playlist ACL roles for the correct visibility
representation in frontend and series creators for some
additional metadata display.

owi92 committed:
Rewrite manage table from table layout to list layout

..and add new filtering options.
This is the core UI overhaul of this PR, which has the goal
of adapting a similar layout to the list style we use for
series and playlist blocks, and also somewhat similar to
the results of the main search. The previous layout didn't
work for mobile at all, but this new one has been built with
extra consideration for smaller screens. Quite a few components
were added. shuffled around, rewritten or repurposed in an attempt
to limit code duplication and make the new table view look good and
work for videos, series and playlists.

LukasKalbertodt commented on frontend/src/layout/header/Search.tsx:
```diff
@@ -132,13 +136,13 @@ export const SearchInput: React.FC<SearchInputProps> = ({
                 position: "absolute",
                 height: "100%",
                 left: 11,
                 fontSize: 23,
                 color: COLORS.neutral60,
-            }} />
```
Small nit: most of these CSS properties are implementation detail of this component. By accepting and rendering a `ReactNode` as `icon`, users of this component are forced to duplicate these CSS declarations. It would be nice to somehow also apply the CSS to the `icon` if its passed. Not sure whats the best way to do it.

owi92 replied:
It can be passed as `IconType` with a few style overrides instead. That keeps the shared css props local.

LukasKalbertodt commented on frontend/src/ui/Blocks/VideoList.tsx:
```diff
@@ -716,7 +717,6 @@ const MenuItem = React.forwardRef<HTMLButtonElement, MenuItemProps>(({
                     ...focusStyle({ inset: true }),
                     "&[disabled]": {
                         fontWeight: "bold",
-                        color: COLORS.neutral80,
```
This change means that for the videolist block layout/sort menus, the disabled (i.e. currently active) item has a different color. Is that change intended? I can't even tell which i like better... just wanted to make you aware of it.

<img width="602" height="558" alt="Image" src="https://github.com/user-attachments/assets/7f6f2fa4-eb54-475a-93dc-10cb01f4bb72" />

<img width="602" height="558" alt="Image" src="https://github.com/user-attachments/assets/ea1ae367-b54b-4f0d-91b7-a65dbca99527" />

owi92 replied:
> Is that change intended?

It is and was done in reaction to a suggestion you made.. like 3 months ago. Quoting:
> Is the selected font less prominent (dark mode: darker) than the non-selected one? They should have the same font color.

I do realize that they actually still don't have the same font color, but at least it's not darker anymore.

Edit: It's now about half a year ago. I still have your suggestions in a file or two though if you're interested.
Anyway, **TLDR**: I'd just keep it as is now.

LukasKalbertodt commented on frontend/src/util/index.tsx:
```diff
@@ -382,3 +385,62 @@ export function ConditionalWrapper({
+            <WithTooltip placement="bottom" tooltip={<>{t(tooltipKey)}</>}>
+                <div css={{
+                    color: "#ddd",
+                    filter: [
+                        "drop-shadow(0 0 1px rgba(0, 0, 0, 0.8))",
+                        "drop-shadow(0 0 1px rgba(0, 0, 0, 0.8))",
+                        "drop-shadow(0 0 3px rgba(0, 0, 0, 0.5))",
+                    ].join(" "),
```
Mhhh have you tried using the same style as the "duration" thing for video thumbnails? This drop shadow works pretty well for almost all situation, but rarely it's still hard to see the icon. Having a plain almost opaque background like with the video duration might work better?

LukasKalbertodt commented on frontend/src/util/index.tsx:
```diff
@@ -382,3 +385,62 @@ export function ConditionalWrapper({
+    return (
+        <div css={{
+            position: "absolute",
+            top: isPlaylist ? 18 : 2,
```
Mhhh this strikes me as a bit weird. This is strongly coupled with the thumbnail component, but it is advertised as individual component. I think this would be better:
- The `Thumbnail` component should have some kind of prop to control whether this is shown
- The `AccessIcon` component can still exist, also as independet thing, but then the outer div (with position absolute) should be removed from that. That positioning stuff is part of the thumbnail logic. The `AccessIcon` can just return an icon with tooltip.

LukasKalbertodt commented on frontend/src/util/index.tsx:
```diff
@@ -382,3 +385,62 @@ export function ConditionalWrapper({
+export const AccessIcon: React.FC<AccessProps> = ({ item, isPlaylist }) => {
+    const { t } = useTranslation();
+    const user = useUser();
+    if (!isRealUser(user)) {
+        return null;
+    }
```
I think it's fine to remove that `if`. The `user` object is not used later. And returning `null` is likely unexpected from a user of this component.

LukasKalbertodt reviewed (changes requested):
Only reviewed a small part of this PR, but wanted to post it already before the weekend. More next week.

LukasKalbertodt commented on backend/src/api/model/shared.rs:
```diff
@@ -146,6 +147,16 @@ impl SortDirection {
 
+#[derive(Debug, Clone, Copy, PartialEq, Eq, GraphQLEnum)]
+pub(crate) enum Visibility {
+    // Accessible to anonymous users (`ROLE_ANONYMOUS` in `read_roles`).
+    Public,
+    // Not public but read roles include at least one non-admin role.
+    Shared,
+    // Only readable for self and admin users.
+    Private,
```
Meant to use `///` doc comment? The "at least one non-admin role" seems slightly incorrect? That's not exactly what's happening. But will comment below again. EDIT: ok I guess this is not wrong, but I still would say sth like "not public, but can be accessed by at least one other person" or sth like that?

LukasKalbertodt commented on backend/src/api/model/shared.rs:
```diff
@@ -206,31 +218,71 @@ where
+        ),
+        Some(Visibility::Shared) => format!("\
+            and not ('ROLE_ANONYMOUS' = any({table}.read_roles)) \
+            and array_length({table}.read_roles, 1) > 1\
+        "),
+        Some(Visibility::Private) => format!("\
+            and not ('ROLE_ANONYMOUS' = any({table}.read_roles)) \
+            and coalesce(array_length({table}.read_roles, 1), 0) <= 1\
```
Either the `coalesce` needs to be in both plcaes or we could use `cardinality`, which you once showed me and it seems like its just better because of the lack of special case?

LukasKalbertodt commented on backend/src/api/model/shared.rs:
```diff
@@ -206,31 +218,71 @@ where
+            "and 'ROLE_ANONYMOUS' = any({table}.read_roles)"
+        ),
+        Some(Visibility::Shared) => format!("\
+            and not ('ROLE_ANONYMOUS' = any({table}.read_roles)) \
+            and array_length({table}.read_roles, 1) > 1\
```
Mhhhh, I'm not yet sure this is the test we want. Yes, make sure ROLE_ANONYMOUS isn't in there. Okay, our documentation mentions that we require Opencast to always have a `read` entry if people have a `write` entry as well. Mh actually yes no, maybe this is correct. Sorry, needed to convince myself.

LukasKalbertodt commented on frontend/src/i18n/locales/de.yaml:
```diff
@@ -582,13 +587,38 @@ manage:
+      visibility-private: Privat
+      visibility-private-tooltip: Kann nur von Ihnen gesehen werden
+      visibility-shared: Geteilt
+      visibility-shared-tooltip: Kann von einer ausgewählten Gruppe gesehen werden
```
Mh this heavily implies singular, right? What about "Kann von ausgewählten Personen/Gruppen gesehen werden"?

LukasKalbertodt reviewed (changes requested):
"Only" the last commit is missing. I only just noticed we have new convenience date selection options, those look good!

owi92 committed:
Remove `hostRealms` from `SeriesManageQuery`

Probably a leftover. Not needed here.

owi92 committed:
Limit need for icon style duplication in repurposed search bar

owi92 committed:
Refactor access icon for general usage

This was strongly tied to thumbnails (and is in fact only used
on them for now) but that required very specific positioning rules
in order to work for both single thumbnails and thumbnail stacks.
This refactor changes it so the thumbnail itself does the positioning
and the icon can be used in other contexts as well.

owi92 committed:
Fix tooltip clipping of access icon in thumbnails

The previous access icon refactor lead to the tooltip being clipped,
so this needed some seperation between tooltip/access div
and actual thumbnail div, so one of those could have `overflow: hidden`
and the other `overflow: visible`.

owi92 committed:
Change access icon background and move some styles

...to make them reusable.
Also switches from non-working (for whatever reason)
container query to a regular ass screen width one.

LukasKalbertodt commented:
Alrighty! This PR might hold the record for "open for the longest time" :sweat_smile: 

We decided to just merge this now. We have gone through a ton of iteration and improvements and the result is really good and way better than before. I still haven't reviewed the main commit and maybe there are some UI/UX things we still need to improve. But I think it's best for the project to just merge it and see from there.
